### DISCUSSION TITLE
Thoughts about develop branch :)

### DISCUSSION THREAD

**Discussion by Natenom** [razonyang/hugo-theme-bootstrap#559] [General]
With your help I managed to build with develop branch. Thank you :)

Here are my thoughts about the changes:

## New "top bar"(?)
I like the new top bar with icons for dark/light/auto mode, color and language and the rearrangement of the parts. Didn't like the old sidebar, which was very slow in Firefox (but fast in Chromium).

## Tags/categories in Post metadata
In the old theme there is one dot between "Reading time" and categories/tags list. In the new theme there is a dot between every single entry in the list, is that a bug?

Old: 

![image](https://user-images.githubusercontent.com/385995/174085620-ac463620-cea6-478d-ba41-f9d886bb61c1.png)

New:

![image](https://user-images.githubusercontent.com/385995/174085580-08da1080-e14e-4958-b2d3-984b1ef88abf.png)

## Actions
Could you please add an option to disable the action button?

(Could probably just disable it myself with css?)

## Font size selector
1. The mouse cursor changes when hovering the entries. Extra small is text cursor, next is normal, then text cursor, normal and "Extra Large" is again text cursor. Tested in Firefox and Chromium.

2. Use "Default" instead of "Medium"?

## Language selector
Is it possible to hide the language selector if there is no translation available? Or hide it via configuration for single language websites?

## Sidebar – Tags and Categories
1. Thanks for more contrast in the counters. But I think they are bit too prominent. I like them to stay more in the background while reading a blog post.
2. The background for categories and tags does not use the selected color from palette. Miss or on purpose?

**Comment by razonyang** [comment]
1. Fix font size selector cursor issue. It's no need to change Medium to Default, since the medium is selected by default.
2. I'll provide a parameter to disable the action's panel later.
3. The post taxonomies buttons of post meta is intended, but it looks a litter superfluous, removed.
4. Hided language selector on single site.
5. As for the sidebar taxonomies, I use the secondary color not primary color, in order to make it lighter(same purpose). I've added a opacity property for those buttons for making them much lighter now.

You can test it online or git pull to your local now.

**Comment by razonyang** [comment]
FYI, the `layouts/archive.html` was removed. You'll need to delete the `layout: archive` from your archives pages.

**Reply by Natenom** [reply]
My old "archive" was always located under "/archiv/" so thats okay for now. But the new "/archive" is not available yet.

**Reply by razonyang** [reply]
@Natenom In this case, you'll still need to specify the `layout = archives` in front matter, since Hugo could not find the `archiv` layout.

**Reply by Natenom** [reply]
Did that, result:

Error: Error building site: failed to render pages: render of "section" failed: execute of template failed: template: _default/archives.html:14:12: executing "content" at <partial "archives/year" (dict "page" . "pages" $pages "year" (index $date 0))>: error calling partial: "/blog/themes/hugo-theme-bootstrap/layouts/partials/archives/year.html:2:56": execute of template failed: template: partials/archives/year.html:2:56: executing "partials/archives/year.html" at <time.AsTime>: error calling AsTime: unable to parse date: http:-01-01 00:00:00
Built in 18627 ms

**Reply by razonyang** [reply]
@Natenom Have you upgrade to the latest commit? The issue should be fixed on single issue via commit b8414c0f2385b9777398fba44fbee09e03ff154a

**Reply by Natenom** [reply]
Jupp, that commit is in the git log.

**Reply by razonyang** [reply]
> Jupp, that commit is in the git log.

How does your archive structure look like? `tree content/archiv`.

**Reply by Natenom** [reply]
```
content/
 pages/
    archiv/
      _index.md
```

_index.md:
```
---
layout: archives
title: Archiv
url: /archiv/
---
```

**Reply by razonyang** [reply]
Just add the `archive.basePath` parameter, you'll need to upgrade theme and change your structure like this:

```toml
// config.toml
[archive]
  basePath = "/archiv"
```

```
$ tree content/archiv/
├── 2022
│   └── _index.md
└── _index.md
```

And then remove the `url` from your front matter. The `.GetPage` function seems doesn't work well with `url` parameter.

**Reply by Natenom** [reply]
That worked. Thank you. But I needed to remove the 2022 directory in order to get some posts from 2022 in the archive.

Oki, so the year is only linked if there is another 2022/_index.md file ... so I need to create files for 2010 - 2022 :)

**Reply by razonyang** [reply]
> That worked. Thank you. But I needed to remove the 2022 directory in order to get some posts from 2022 in the archive.
> 
> And the years are not linked.

the years directory/section is required if you need the year likable...

**Comment by Natenom** [comment]
1. Medium is selected by default but what if you selected another one and want go back to default and don't know which that was?
2. Okay :)
3. Better. Thank you.
4. Works.
5. The hover behaviour is wired now. First, background is lighter on hover and then instantly changes to dark again, like an animation.

**Comment by razonyang** [comment]
1. I'll keep the medium, they look equal to me... But you still can change it by creating your own i18n translations to your project root: https://github.com/razonyang/hugo-theme-bootstrap/blob/edcd44e14e4dfca49a46f1ee37ca5294b73af1dc/i18n/en.toml#L107 
5. The taxonomy's buttons' opacity was reset to `1` on hover, nothing more, it's reasonable to me.

**Reply by Natenom** [reply]
1. Okay.
2. Don't understand it yet. Will ask my friend who knows about css :)

**Comment by Natenom** [comment]
The mouse cursor in "Dark/Light/Auto" is different (default cursor) from the other menus (hand).

**Reply by razonyang** [reply]
Fixed by 87f21868ebedc8d080b377816fd993a64a2ad65c. Thank you.
